src/main/java/roomescape/domain/entity/Reservation.java

```diff
@@ -9,12 +12,32 @@ public class Reservation {
     private final Time reservationTime;
 
     public Reservation(Long id, String name, LocalDate reservationDate, Time reservationTime) {
+        validate(name, reservationDate, reservationTime);
         this.id = id;
         this.name = name;
         this.reservationDate = reservationDate;
         this.reservationTime = reservationTime;
     }
 
+    private void validate(String name, LocalDate reservationDate, Time reservationTime) {
+        if (name == null || name.isBlank()) {
+            throw new InvalidReservationRequestException("이름은 필수 입력 값입니다.");
+        }
+
+        if (reservationDate == null) {
+            throw new InvalidReservationRequestException("예약 날짜는 필수 입력 값입니다.");
+        }
+
+        if (reservationTime == null) {
+            throw new InvalidReservationRequestException("예약 시간은 필수 입력 값입니다.");
+        }
+
+        if (reservationDate.isBefore(LocalDate.now()) ||
+                (reservationDate.isEqual(LocalDate.now()) && reservationTime.getTime().isBefore(LocalTime.now()))) {
+            throw new InvalidReservationRequestException("과거로는 예약할 수 없습니다.");
+        }
```

Comment:
디미터 법칙에 대해서 알아보시고 적용시켜보시죠!!


```suggestion

        if (reservationDate.isBefore(LocalDate.now()) ||
                (reservationDate.isEqual(LocalDate.now()) && reservationTime.isBefore(LocalTime.now()))) {
            throw new InvalidReservationRequestException("과거로는 예약할 수 없습니다.");
        }
```

Reply:
확실히 Time의 관심사 인 것 같습니다! 수정했습니다